curl_request.hpp

```diff
@@ -94,6 +95,27 @@ class Request
         curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, method.c_str());
         return this->execute();
     }
+    /**
+     * @brief Download a file from the specified URL and save it to the given file path
+     * @param file_path The path where the downloaded file should be saved
+     * @return The result of the cURL request
+     */
+    int download_file(const std::string& file_path)
```

Comment:
Method needs to outline that it will perform a GET request, and set the appropriate HTTPGET option